test/test_autograd.py

```diff
@@ -5467,6 +5467,28 @@ def test_input_buffer_accum(self):
         # Given gradients should not be modified inplace
         self.assertEqual(grad_out1, grad_out1_original)
 
+    def test_no_unnecessary_unwrapping(self):
+        a = torch.randn(5, requires_grad=True)
+        a_orig = a.detach().clone()
+        b = a * a
+        c = a * b
```

Comment:
nit: if you want to test an output here to make sure it is properly unwrap, you can use exp() and make sure the `_saved_result` is not the same as the returned Tensor.